bus/api.go

```diff
@@ -190,3 +190,11 @@ type MigrationContract struct {
 	HostIP  string               `json:"hostIP"`
 	ID      types.FileContractID `json:"id"`
 }
+
+// UploadParams contains the metadata needed by a worker to upload an object.
+type UploadParams struct {
```

Comment:
Slightly different than what I had in mind, I figured the worker would be using contract sets. the `min` and `total` shards would be something that's optionally provided by the user but defaults to 10-30 on the upload endpoints? 

This means i would have to change the bus config to upload with a different redundancy? This also means the bus has to having some logic behind the upload params route which is actually useful for acquiring a lock and the likes.

Not against it at all i was just wondering about the non-flexible redundancy. Also just in general it's not really how I thought we'd use the contract sets.

Reply:
The redundancy is also part of the autopilot config, so I think we need to expose this is parameters on the upload object endpoint rather than include it in the `UploadParams`. The worker could get the `Contracts` by fetching a contract set with some predefined name like `goodForUpload`, and we already have an endpoint on the bus for `blockheight`.

Reply:
Sure, we can allow overriding the redundancy settings. I'll wait to add that until it's needed, though.

The bus should store the default redundancy. If the autopilot is currently storing the redundancy settings, we should change it to fetch the settings from the bus.